controlplane/internal/controller/clusterdeployment_controller.go

```diff
@@ -67,42 +68,30 @@ func (r *ClusterDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		}
 		return ctrl.Result{}, err
 	}
-	log.Info("Reconciling ClusterDeployment", "name", clusterDeployment.Name, "namespace", clusterDeployment.Namespace)
-
-	agentCPList := &v1alpha1.AgentControlPlaneList{}
-	if err := r.Client.List(ctx, agentCPList, client.InNamespace(clusterDeployment.Namespace)); err != nil {
-		if apierrors.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
-		return ctrl.Result{}, err
-	}
-	log.Info("found agentcontrolplane", "num", len(agentCPList.Items))
+	log.WithValues("ClusterDeployment Name", clusterDeployment.Name, "ClusterDeployment Namespace", clusterDeployment.Namespace)
+	log.V(logutil.TraceLevel).Info("Reconciling ClusterDeployment")
 
-	if len(agentCPList.Items) == 0 {
+	if len(clusterDeployment.GetOwnerReferences()) < 1 {
+		log.V(logutil.DebugLevel).Info("ClusterDeployment is not owned by AgentControlPlane")
 		return ctrl.Result{}, nil
 	}
 
-	for _, acp := range agentCPList.Items {
-		if IsAgentControlPlaneReferencingClusterDeployment(acp, clusterDeployment) {
-			log.Info("ClusterDeployment is referenced by AgentControlPlane")
-			return r.ensureAgentClusterInstall(ctx, clusterDeployment, acp)
-		}
+	acp := controlplanev1alpha1.AgentControlPlane{}
+	if err := util.GetTypedOwner(ctx, r.Client, clusterDeployment, &acp); err != nil {
```

Comment:
Wouldn't this cover also the previous condition? Would we care enough to have a specific condition if not enough references?

Reply:
The GetTypedOwner returns an error if there's no owners, which would cause this ClusterDeployment to be reconciled again. 

I think we should ignore any ClusterDeployment (not reconcile again) if there are no owners. What do you think?

Reply:
What's the difference between "there is no owner" and "there are owners, just not of the type you are looking for"? Wouldn't we need to reconcile in the same way? (Either requeue or not requeue on both?)

Reply:
That's true. What I want to avoid is requeueing a clusterdeployment that doesn't belong to us (cluster api agent). I'll change this to reflect that